logverification/app/appentry.go

```diff
@@ -0,0 +1,292 @@
+	"github.com/datatrails/go-datatrails-common/logger"
+	"github.com/datatrails/go-datatrails-merklelog/massifs"
+	"github.com/datatrails/go-datatrails-merklelog/mmr"
+	"github.com/google/uuid"
+)
+
+/**
+ * AppEntry is the app provided data for a corresponding log entry.
+ * An AppEntry will derive fields used for log entry inclusion verification.
+ *
+ * The format for an MMR entry is the following:
+ *
+ * H( Domain | MMR Salt | Serialized Bytes)
+ *
+ * Where:
+ *   * Domain           - the hashing schema for the MMR Entry
+ *   * MMR Salt         - datatrails provided fields included in the MMR Entry (can be found in the corresponding Trie Value on the log)
+ *   * Serialized Bytes - app (customer) provided fields in the MMR Entry, serialized in a consistent way.
+ *
+ *
+ * The format for a Trie Entry is the following:
+ *
+ * ( Trie Key | Trie Value )
+ *
+ * Where the Trie Key is:
+ *
+ * H( Domain | LogId | AppId )
+ *
+ * And Trie Value is:
+ *
+ * ( Extra Bytes | IdTimestamp )
+ */
+
+const (
+	MMRSaltSize = 32
+
+	IDTimestapSizeBytes = 8
+)
+
+// MMREntryFields are the fields that when hashed result in the MMR Entry
+type MMREntryFields struct {
+
+	// Domain defines the hashing schema for the MMR Entry
+	Domain byte
+
+	// SerializedBytes are app (customer) provided fields in the MMR Entry, serialized in a consistent way.
+	SerializedBytes []byte
+}
+
+// AppEntry is the app provided data for a corresponding log entry.
+//
+// It contains key information for verifying inclusion of the corresponding log entry.
+//
+// NOTE: all fields are sourced from the app data, or derived from it.
+// NONE of the fields in an AppEntry are sourced from the log.
+type AppEntry struct {
+	// AppId is an identifier of the app committing the merkle log entry
+	AppId string
+
+	// LogId is a uuid in byte form of the specific log identifier
+	LogId []byte
+
+	// ExtraBytes are extrabytes provided by datatrails for the specific app
+	ExtraBytes []byte
+
+	// MMREntryFields used to determine the MMR Entry
+	MMREntryFields *MMREntryFields
+
+	// MerkleLogCommit used to define information about the log entry
+	MerkleLogCommit *assets.MerkleLogCommit
+
+	// MerkleLogConfirm used to define information about the log seal
+	MerkleLogConfirm *assets.MerkleLogConfirm
+}
+
+// NewAppEntry creates a new app entry entry
+func NewAppEntry(
+	appId string,
+	logId []byte,
+	extraBytes []byte,
+	mmrEntryFields *MMREntryFields,
+	merklelogCommit *assets.MerkleLogCommit,
+) *AppEntry {
+
+	verifiableLogEntry := &AppEntry{
+		AppId:           appId,
+		LogId:           logId,
+		ExtraBytes:      extraBytes,
+		MMREntryFields:  mmrEntryFields,
+		MerkleLogCommit: merklelogCommit,
+	}
+
+	return verifiableLogEntry
+}
+
+// MMREntry derives the mmr entry of the corresponding log entry from the app data.
+//
+// MMREntry is:
+//   - H( Domain | MMR Salt | Serialized Bytes)
+func (ae *AppEntry) MMREntry() ([]byte, error) {
```

Comment:
There's always the æ ligature if we want to save a character /s

Reply:
love it!